scene/resources/shader.cpp

```diff
@@ -248,6 +257,9 @@ void Shader::_bind_methods() {
 
 	ClassDB::bind_method(D_METHOD("get_shader_uniform_list", "get_groups"), &Shader::_get_shader_uniform_list, DEFVAL(false));
 
+	ClassDB::bind_method(D_METHOD("inspect_native_shader_code"), &Shader::inspect_native_shader_code);
+	ClassDB::set_method_flags(get_class_static(), _scs_create("inspect_native_shader_code"), METHOD_FLAGS_DEFAULT | METHOD_FLAG_EDITOR);
```

Comment:
Is there a strong use case for exposing this to the scripting API? You should be able to use this within the editor without exposing it. It's only needed to expose it for editor plugins, but I'm not sure why they'd need to programmatically spawn the dialog.

Reply:
I copied the existing implementation on Material. I assumed there was some reason it was done the way it is:
https://github.com/godotengine/godot/blob/0a4aedb36065f66fc7e99cb2e6de3e55242f9dfb/scene/resources/material.cpp#L150-L151

Reply:
That looked strange to me too, but I don't really understand how those flags work. I guess "DEFAULT" allows it to be called from scripting, but "EDITOR" limits that to an editor build?

At any rate, its fine since this is indeed copy-pasted from the existing implementation

Reply:
And, to state the maybe not-so-obvious - there's no good way I saw for the `Material` implementation call the `Shader` implementation, since the base material class only has a shader RID reference, not a Shader resource reference.

Reply:
Ah, looks like @clayjohn you voiced concerns in the original PR https://github.com/godotengine/godot/pull/44950. This is a niche enough feature that I'm not sure if it's worth putting a lot of time into. But I want to at least take a cursory look at possible improvements before merging this as-is.

Reply:
Okay, this isn't flawless proof, but I [searched across all of github](https://github.com/search?q=inspect_native_shader_code&type=code) for usage of "inspect_native_shader_code", and it is 99% either forks of godot repos, or godot bindings in various languages. There is one instance of user code, in a [fuzzer for godot](https://github.com/qarmin/Qarminer/blob/cebb2e69733fcada7324f58232912a4383f40340/Autoloads/BasicData.gd#L176). I could not find any instances where a plugin with public source on github makes use of `Material.inspect_native_shader_code()` being exposed to scripting.

I don't know how high the bar is for unexposing a function to scripting is in a minor version.

Reply:
I looked into it more, and the method needs to be added to the class DB (and exposed to scripting) in order for it to show up in the inspector of the resource. If we were *only* adding it to the shader editor, there would be no need to expose it.

See the inspector dock implementation: https://github.com/godotengine/godot/blob/842f98239713fd10cfd648cd6aa3781895f289eb/editor/inspector_dock.cpp#L566-L583